debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/RecordMakers.java

```diff
@@ -173,7 +173,7 @@ public boolean assign(long tableNumber, TableId id) {
         TableSchema tableSchema = schema.schemaFor(id);
         if (tableSchema == null) return false;
 
-        String topicName = topicSelector.getTopic(id);
+        String topicName = tableSchema.getParentSchemaName();
```

Comment:
Really, the proper thing to do is to have a single envelope schema that essentially for the `before` and `after` field schemas a union of all of the table-specific `Schema` instances. But this isn't easy, and to build that schema we'd want to know all the tables up front so that the schema is right before we produce our first record. And, I'm not sure whether Kafka Connect `Schema` can handle that; might be a question for Ewen.

Anything else really creates a lot of churn in the Schema Registry. Need to think a bit more on this.